### PR TITLE
add a Crispy Doom automap color scheme

### PR DESCRIPTION
**Pull request by fabiangreffrath**
https://www.doomworld.com/forum/topic/112333-this-is-woof-1450-apr-30-2024/?page=92&tab=comments#comment-2804787

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
I need someone to double-check the color indices with Crispy Doom, please.

**Comment by fabiangreffrath** [comment]
Should we disable blinking keys for the Vanilla scheme again? First, it*s not very Vanilla at all and second, it*s pretty useless for red and yellow doors anyway.

**Comment by rfomin** [comment]
> Should we disable blinking keys for the Vanilla scheme again? First, it_s not very Vanilla at all and second, it_s pretty useless for red and yellow doors anyway.

I think it's confuding that "Colour Preset" will change other options. Then maybe we should rename it to "Automap Preset".

> I need someone to double-check the color indices with Crispy Doom, please.

Seems right.

File: src/am_map.c [hunk 1]
```diff
@@ -2376,7 +2384,8 @@ void AM_BindAutomapVariables(void)
   M_BindBool("map_smooth_lines", &map_smooth_lines, NULL, true, ss_auto,
              wad_no, "Smooth automap lines");
 
-  M_BindNum("mapcolor_preset", &mapcolor_preset, NULL, 1, 0, 2, ss_auto, wad_no,
+  M_BindNum("mapcolor_preset", &mapcolor_preset, NULL, AM_PRESET_BOOM,
+            AM_PRESET_VANILLA, AM_PRESET_ZDOOM, ss_auto, wad_no,
             "Automap color preset (0 = Vanilla Doom; 1 = Boom; 2 = ZDoom)");
```
Comment (ceski-1) [inline src/am_map.c:2389]: Don't forget to update the description.

**Comment by fabiangreffrath** [comment]
> I think it's confusing that "Colour Preset" will change other options. Then maybe we should rename it to "Automap Preset".

Right, we'll keep it as it is. After all, people complained about the visibility of the feature, not its availability. 

> Seems right.

Thanks for checking!

File: src/hu_stuff.c [hunk 2]
```diff
@@ -694,7 +695,18 @@ static void HU_widget_build_title (void)
 
   M_StringConcat(hud_titlestr, s, sizeof(hud_titlestr));
 
-  HUlib_add_string_to_cur_line(&w_title, hud_titlestr);
+  // [crispy] print the map title in white from the first colon onward
+  if (mapcolor_preset == AM_PRESET_CRISPY ||
+      mapcolor_preset == AM_PRESET_ZDOOM)
+  {
+    s = M_StringReplace(hud_titlestr, ":", (const char[]){ ':', '\x1b', '0'+CR_GRAY, '\0' });
```
Comment (rfomin) [inline src/hu_stuff.c:702]: Crispy has this feature? In GZDoom the first part of the title is red.
It's too late to remove the `hud_titlestr` colouring completely, isn't it? 😄

Comment (fabiangreffrath) [reply src/hu_stuff.c:702]: > Crispy has this feature? In GZDoom the first part of the title is red. 

Nit: it's unmodified, so e.g. in btsx_e1 it remains in cyan. 

> It's too late to remove the `hud_titlestr` colouring completely, isn't it? 😄

I'm afraid yes. 😉

Comment (rfomin) [reply src/hu_stuff.c:702]: I vote to remove the extra colouring of the map title.

Comment (fabiangreffrath) [reply src/hu_stuff.c:702]: You mean the bi-coloring? Okay, I can `#if 0` it out, maybe forks want to keep it.

Comment (rfomin) [reply src/hu_stuff.c:702]: > You mean the bi-coloring? Okay, I can `#if 0` it out, maybe forks want to keep it.

Yes please. I want to use Crispy preset without this.

**Review by rfomin — APPROVED** [review]
Thank you.

**Pull request merged**